### PR TITLE
CosmosDB: Add missing implementations for creating other data models

### PR DESCRIPTION
**Pull request by milismsft**
Add missing implementations to create Cassandra, Azure Table and Gremlin data model CosmosDB accounts.

### CODE REVIEW

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java [hunk 1]
```diff
@@ -154,6 +162,61 @@
...
+            WithConsistencyPolicy withDataModelCassandra();
+
+            /**
+             * Creates an Azure Table CosmosDB account.
+             *
+             * @return the next stage of the definition
+             */
+            @Method
+            @Beta(SinceVersion.V1_10_0)
+            WithConsistencyPolicy withDataModelAzureTable();
```
Comment (hovsepm) [inline azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: >withDataModelAzureTable [](start = 34, length = 23)

Should it be just withDataModel method and then a parameter of Enum type with all the supported DB Types and enum members? Too many methods in Intelisence that start with the same prefix (A long one in this case) creates a bunch of clutter IMHO

Comment (hovsepm) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: also it is kind of weird construct `withDataModelSql` . More natural would be `withSqlAccount` or `withSqlTyoe` but in that case you will have types in a hard to find order in the intelisence. So reiterating my proposal to have one method and bunch of enum values :)

Comment (JREndeanMSFT) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: In the management portal it is called API 
cosmosdbportalui

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/implementation/CosmosDBAccountImpl.java [hunk 2]
```diff
@@ -162,6 +173,52 @@ public CosmosDBAccountImpl withKind(DatabaseAccountKind kind) {
...
+    public CosmosDBAccountImpl withKind(DatabaseAccountKind kind, Capability... capabilities) {
+        this.inner().withKind(kind);
+        this.inner().withCapabilities(Arrays.asList(capabilities));
+        return this;
+    }
+
+    @Override
+    public CosmosDBAccountImpl withDataModelSql() {
+        this.inner().withKind(DatabaseAccountKind.GLOBAL_DOCUMENT_DB);
+        return this;
```
Comment (JREndeanMSFT) [inline azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/implementation/CosmosDBAccountImpl.java]: Are you sure the "defaultExperience" tags are not needed on these? If you create each API type in the portal, they all have a "defaultExperience" tag.

Comment (milismsft) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/implementation/CosmosDBAccountImpl.java]: The tags do not impact resource provisioning; if they do then it's a bug. While the tests I've added set a generic tag it's not required to create an CosmosDB account.

File: azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java (hunk 1 above)
Comment (milismsft) [reply azure-mgmt-cosmosdb/src/main/java/com/microsoft/azure/management/cosmosdb/CosmosDBAccount.java]: We do not need to match exactly what the Azure Portal does. Data model fits better the type of the account (see the explanation behind the little "i").

There's already a method that takes enums (see the withKind() overload I just added). These new methods are direct shortcuts for creating the respective CosmosDB account.

**Pull request merged**